### DISCUSSION TITLE
Make WT hard to close (disable the close button)?

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#14988] [General]
If I remove SC_CLOSE from a console's system menu, the close button (X) becomes disabled (greyed).  I can remove SC_CLOSE from the CASCADIA_HOSTING_WINDOW_CLASS window's system menu but that doesn't disable the button.  Is it possible to disable WT's close button?  If so, suggestions are welcomed.

**Comment by bzzrak** [comment]
try going to settings -> appearance and disabling "hide the title bar".

**Reply by vefatica** [reply]
That gives a different Close button (X) to disable.  I didn't try it but I imagine that one would be disabled if I removed SC_CLOSE from the system menu.  But all that leaves WT looking rather ugly.  I wouldn't want it like that all the time since I only rarely want to disable closing it (when I'm monitoring something or waiting for something).

I have one solution that I also use with OutLook, namely, inject a DLL which uses a WH_GETMESSAGE hook to turn SC_CLOSE into SC_MINIMIZE.

**Reply by bzzrak** [reply]
Yeah it's not exactly aesthetic. But if you rely on it, I think you could get used to it.

**Comment by zadjii-msft** [comment]
You can't get rid of the close button specifically, but you _may_ be interested in [Focus mode](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/actions#toggle-focus-mode) or even [Read only mode](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/actions#mark-a-pane-as-read-only)

**Reply by vefatica** [reply]
I rather like WT in its normal mode.  I don't really care about getting rid of the button; I just want to disable it.  As I said, I can do that with a tiny (6K) DLL and a WH_GETMESSGE hook.  I'm looking for a better/easier way.
